### PR TITLE
Find View UI Improvements

### PR DESCRIPTION
**Pull request by altair1673**
### Prerequisites

- [X] Reviewed the [checklist](CHECKLIST.md)

- [ ] Reviewed feedback from the "Sonar Cloud" bot. Note that you have to wait
    for the "CI / Unit Tests") to complete first. Failed Unit tests can be 
    debugged by adding the label "verbose logging" to the GitHub PR.

### Description of the Change
Changed title from "Find View" to "Find". In the "Basic Find" tab, "In Attributes" label has been updated to "Search Attributes". All attributes are selected by default. 
In the "Advanced Find" tab the number of matches found is displayed.
The "Match Criteria" and "Current Selection" labels have been spaced out.
Updated the "Ignore Case" label in advanced find tab.
Decided not to update strings in the "Look for" drop down to plural as it would impact code in other areas of the codebase.

### Why Should This Be In Core?
Improves user experience.

### Benefits
Improves readability of UI text elements. Adds in features that were available in the old find view which improves experience.

### Verification Process
Unit tests pass, UI is readable and well spaced out.

### Applicable Issues
#1692

### CODE REVIEW

File: CoreFindView/src/au/gov/asd/tac/constellation/views/find2/FindViewController.java [hunk 1]
```diff
@@ -282,7 +286,8 @@ public void replaceMatchingElements(final boolean replaceAll, final boolean repl
     }
 
     public void retrieveAdvancedSearch(final boolean findAll, final boolean findNext) {
-        final AdvancedSearchPlugin advancedSearchPlugin = new AdvancedSearchPlugin(currentAdvancedSearchParameters, findAll, findNext);
+
+        AdvancedSearchPlugin advancedSearchPlugin = new AdvancedSearchPlugin(currentAdvancedSearchParameters, findAll, findNext);
```
Comment (antares1470) [inline CoreFindView/src/au/gov/asd/tac/constellation/views/find2/FindViewController.java:290]: can you make this final?

File: CoreFindView/src/au/gov/asd/tac/constellation/views/find2/FindViewController.java [hunk 2]
```diff
@@ -340,4 +351,20 @@ public AdvancedSearchParameters getCurrentAdvancedSearchParameters() {
...
+     * @return
+     */
+    public IntegerProperty getNumResultsFound() {
+        return numResultsFoundFlag;
+    }
+
+    /**
+     * Sets amount of results found by advanced search
+     */
+    public void setNumResultsFound(int value) {
```
Comment (antares1470) [inline CoreFindView/src/au/gov/asd/tac/constellation/views/find2/FindViewController.java:366]: are you able to make the parameter final?

File: CoreFindView/src/au/gov/asd/tac/constellation/views/find2/plugins/advanced/AdvancedSearchPlugin.java [hunk 3]
```diff
@@ -279,6 +286,10 @@ protected void edit(GraphWriteMethods graph, PluginInteraction interaction, Plug
         }
         findAllMatchingResultsList.addAll(findResultSet);
 
+        final int resultsFoundSize = findAllMatchingResultsList.size();
+
+
+
```
Comment (antares1470) [inline CoreFindView/src/au/gov/asd/tac/constellation/views/find2/plugins/advanced/AdvancedSearchPlugin.java:292]: are you able to remove some of the extra newlines here?

File: CoreFindView/src/au/gov/asd/tac/constellation/views/find2/plugins/advanced/AdvancedSearchPlugin.java [hunk 4]
```diff
@@ -128,6 +133,7 @@ protected void edit(GraphWriteMethods graph, PluginInteraction interaction, Plug
         final int stateId = FindViewConcept.MetaAttribute.FINDVIEW_STATE.ensure(graph);
         FindResultsList foundResult = graph.getObjectValue(stateId, 0);
 
+        LOGGER.log(Level.SEVERE, "Running edit method");
```
Comment (antares1470) [inline CoreFindView/src/au/gov/asd/tac/constellation/views/find2/plugins/advanced/AdvancedSearchPlugin.java:136]: Are you able to remove this log entry?

File: CoreFindView/src/au/gov/asd/tac/constellation/views/find2/plugins/advanced/AdvancedSearchPlugin.java [hunk 5]
```diff
@@ -735,4 +753,7 @@ private int getIndex(final AdvancedSearchParameters currentParameters, final Adv
     public String getName() {
         return "Find: Advanced Search";
     }
+
+
+
```
Comment (antares1470) [inline CoreFindView/src/au/gov/asd/tac/constellation/views/find2/plugins/advanced/AdvancedSearchPlugin.java:758]: similar here

**Review by antares1470 — CHANGES_REQUESTED** [review]
Thanks @altair1673, most of it seems alright in testing. One thing though is that I think the text for the advanced find result count blends in a little too well with the rest of the UI where it currently is. Might be helpful to either move that label below all the different queries (above the Find buttons) or potentially display in the same way the old Find view did (which I think is via the PluginInteraction object)
newFindAdvanced

Also, if you are able to make it such that the label isn't visible at all when there isn't any search, that I think would be a nice inclusion.
newFindAdvanced2

**Review by antares1470 — CHANGES_REQUESTED** [review]
Managed to get this exception which I confirmed isn't occurring in master:
```
java.lang.IllegalArgumentException: Children: duplicate children added: parent = Grid hgap=0.0, vgap=0.0, alignment=TOP_LEFT
	at javafx.graphics/javafx.scene.Parent$3.onProposedChange(Parent.java:558)
	at javafx.base/com.sun.javafx.collections.VetoableListDecorator.add(VetoableListDecorator.java:206)
	at javafx.graphics/javafx.scene.layout.GridPane.add(GridPane.java:974)
	at au.gov.asd.tac.constellation.views.find2.components.AdvancedFindTab.updateButtons(AdvancedFindTab.java:208)
	at au.gov.asd.tac.constellation.views.find2.components.FindViewTabs.lambda$new$0(FindViewTabs.java:52)

```
Reproducer: Open new Find View, switch to Advanced Tab, switch to Basic Tab, switch to Advanced Tab.

I think you might have also missed one of my comments from my previous review (which I've unresolved)

**Review by antares1470 — APPROVED** [review]

**Review by Delphinus8821 — APPROVED** [review]

**Comment by GammaVel** [comment]
Thanks  @altair1673 for making these improvements to the new Find view. They are looking great. Understand that you could not implement the plural for Node, Transaction and Edge but I don't think it is a big issue. 

I have gotten a couple of runtime errors when using the new Find functionality but they are not related to your updates. I will keep following up to see if they can be replicated and if so, raise new issues. 

Thanks again @altair1673 !

**Pull request merged**